NEO-TAT/tat_flutter
# Refactor score page (Only score parts)

## Description 
This PR tried to refactor the score page (also known as the grade page), which specifically did the following things:
1. The score detail info in each semester has been refined into a 3-part-metric:
  - the `course score` section
  - the `semester score` section
  - the `rank` section
 2. The operation in this page, the `refresh` and `calculate credit`, which were originally located in a `PopupMenuButton`, has been redesigned to `IconButton` directly put inside the `AppBar`.
 3. Change the divider color of the theme for using the `Divider` widget in somewhere.

## How to Verify? 
1. Go to the score page, refresh, and see if the score in each semester could be showed properly (no any critical UI problems).
2. Go to course table page, check the background looks the same as before.
3. Go to a course detail page, check the classmates info section's bg looks the same as before.
4. repeat the steps mentioned above with ENG and dark mode.
5. click the `refresh` button in the appbar, check if it works.
6. click the `calculate` button in the appbar, check if it works.
 
## Screenshots/GIF/Test Results (Optional) 

<img width="50%" src="https://user-images.githubusercontent.com/47718989/210593049-d181201b-4560-4d2c-b825-e1c360e30aa8.jpg"><img width="50%" src="https://user-images.githubusercontent.com/47718989/210593078-4c796e69-b9e9-4849-8754-c28fed58e911.jpg">
